src/js/components/AppListComponent.jsx

```diff
@@ -314,22 +366,21 @@ var AppListComponent = React.createClass({
       appsHealthCount: getInitialFilterCounts(HealthStatus)
     };
 
-    // Global search view - only display filtered apps
+    // Global search view - only display filtered apps & groups
     if (this.hasFilters()) {
       appListViewType = AppListViewTypes.APP_LIST;
 
-      nodesSequence = this.filterNodes(lazy(state.apps), filterCounts)
+      nodesSequence = lazy(this.filterNodes(state.apps, filterCounts))
```

Comment:
Hello Felix,
I had to mutate the `nodesSequence` in order to append the group nodes and it was not working as expected when I was using the lazy.js evaluation. This, combined with a comment from Philip, mentioning that the plan is to move away from `lazy.js`, lead me to remove lazy dependencies from the `filterNodes` function. I wrapped it only to preserve the lazy interface of the result.

If I got understood it wrong, an alternative would be:

- On line 246 : Change it back to `.each`
- On line 225 : Change `.filter` to `.reduce( .. , [] )` and in the function body `.push` the accepted items plus the group items  

I can try this when I get to my laptop at home.